eglot-jl.el

```diff
@@ -71,10 +76,31 @@ Otherwise returns nil"
 (cl-defmethod project-roots ((project (head julia)))
   (list (cdr project)))
 
+(defun eglot-jl--sysimage-args ()
+  "Returns a list of command-line switches suitable for sysimage loading.
+Return nil if no system image has been found or if system images
+have been disabled altogether."
+  (when eglot-jl-enable-sysimage
+    (let ((julia-version (with-temp-buffer
+                           (call-process eglot-jl-julia-command nil t nil
+                                         "-e" "print(Base.VERSION)")
+                           (buffer-string))))
+      (car
+       (remove nil
+               (mapcar
```

Comment:
I think this code is much clearer if it simply checks `system-type` and then looks specifically for the appropriate file-extension rather than checking all of them.

Reply:
You're right. Among the options I thought of, my personal favorite would be to implement the logic of the sysimage name construction once, and re-use it both in `compile.jl` and `eglot-jl.el`. Unless I'm missing something, this would be feasible only if the "master implementation" of this logic is in Julia, which increases a bit the overhead of spawning a julia process in order to determine the sysimage name:
```
$ time julia --compile=min -e 'include("utils.jl"); print(sysimage_path())'                                                          
/home/francois/projets/emacs/eglot-jl/eglot-jl.1.5.0-rc1.0.so
real    0m0,256s
user    0m0,288s
sys     0m0,112s
```

However, if we want to reduce the overhead to a minimum, I could of course do as you propose, and re-implement the exact same logic in elisp. In which case we'd better wait to be sure about which information we want to encode in the sysimage name.

All this makes this issue rather intricately linked to other points that you raise, so that our final take on this will probably come naturally from the choices we make on other issues.